src/tool/cppwinrt/cppwinrt/code_writers.h

```diff
@@ -591,35 +591,42 @@ namespace xlang
                 }
 
                 w.write(format, param_name, param_name);
+                continue;
             }
-            else
+
+            TypeDef signature_type;
+            auto category = get_category(param_signature->Type(), &signature_type);
+
+            if (param.Flags().In())
             {
-                if (param.Flags().In())
+                switch (category)
                 {
-                    XLANG_ASSERT(!param.Flags().Out());
-
-                    if (wrap_abi(param_signature->Type()))
-                    {
-                        w.write("get_abi(%)", param_name);
-                    }
-                    else
-                    {
-                        w.write(param_name);
-                    }
+                case param_category::object_type:
+                case param_category::string_type:
+                    w.write("*(void**)(&%)", param_name);
```

Comment:
Replace C style cast

Reply:
Going to leave this as-is for now. It's tricky because this is used for in params so this goes from:

`(void**)(&param)` 

To this:

`reinterpret_cast<void**>(const_cast<FULL_TYPE_NAME_HERE*>(&param))` 

This is obviously super verbose and I'd argue makes the code harder to debug through.